File: WhatIsForDinnerFrontEnd/src/app/saved-recipe.ts
// To parse this data:
//
//   import { Convert } from "./file";
//
//   const savedRecipe = Convert.toSavedRecipe(json);

export interface SavedRecipe {
    id:                   number;
    recipeId:             number;
    title:                string;
    ingredients:          string;
    ingredientAmount:     string;
    image:                string;
    readyInMinutes:       number;
    servings:             number;
    analizedInstructions: string;
    favorites:            any[];
}

// Converts JSON strings to/from your types
export class Convert {
    public static toSavedRecipe(json: string): SavedRecipe {
        return JSON.parse(json);
    }

    public static savedRecipeToJson(value: SavedRecipe): string {
        return JSON.stringify(value);
    }
}

File: WhatIsForDinnerBackEnd/WhatIsForDinnerBackEnd/Models/SavedRecipe.cs
using System;
using System.Collections.Generic;

namespace WhatIsForDinnerBackEnd.Models;

public partial class SavedRecipe
{
    public int Id { get; set; }

    public int RecipeId { get; set; }

    public string? Title { get; set; }

    public string? Ingredients { get; set; }

    public string Image { get; set;}
    public int? ReadyInMinutes { get; set; }

    public int? Servings { get; set; }

    public string? AnalizedInstructions { get; set; }
  

    public virtual ICollection<Favorite> Favorites { get; set; } = new List<Favorite>();
}
